### PR TITLE
Custom deployments in Online Endpoints with MLflow

### PR DESCRIPTION
**Pull request by santiagxf**
# Description

This PR contains an example for serving MLflow models in online endpoints using custom environments.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb [hunk 1]
```diff
@@ -2,43 +2,40 @@
  "cells": [
```
Comment (trangevi) [inline sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb:2]: What is this file and why is it different than the other one you're adding in this PR?

Comment (trangevi) [reply sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb:2]: Assuming that it's just the difference between a custom deploy and NCD, maybe the notebook names should be more descriptive of that?

**Review by trangevi — APPROVED** [review]

File: sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb (hunk 1 above)
Comment (santiagxf) [reply sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb:2]: Are you suggesting to name this one NCD? I don't one to use that name that much cause it is an internal way to name the deployment. That's why I named the other one "with-script" to indicate it is with an scoring script. Should I name it custom?

Comment (wamartin-aml) [reply sdk/python/endpoints/online/mlflow/online-endpoints-deploy-mlflow-model.ipynb:2]: you could name it -no-script directly if that would be helpful for users

File: sdk/python/endpoints/online/mlflow/sklearn-diabetes/environment/conda.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+channels:
+- conda-forge
+dependencies:
+- python=3.7.11
```
Comment (wamartin-aml) [inline sdk/python/endpoints/online/mlflow/sklearn-diabetes/environment/conda.yml:4]: not that important for this PR, but we should probably start upgrading examples to 3.8 soon

File: sdk/python/endpoints/online/mlflow/sklearn-diabetes/src/score.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import json
+import mlflow
+from io import StringIO
+from mlflow.pyfunc.scoring_server import infer_and_parse_json_input, predictions_to_json
+
+
+def init():
+    global model
+    global input_schema
+    # This name is model.id of model that we want to deploy deserialize the model file back
```
Comment (wamartin-aml) [inline sdk/python/endpoints/online/mlflow/sklearn-diabetes/src/score.py:12]: this comment is a bit confusing to read

**Review by wamartin-aml — APPROVED** [review]

File: sdk/python/endpoints/online/mlflow/sklearn-diabetes/environment/conda.yml (hunk 2 above)
Comment (santiagxf) [reply sdk/python/endpoints/online/mlflow/sklearn-diabetes/environment/conda.yml:4]: Indeed Walter, I already started in another PR!

File: sdk/python/endpoints/online/mlflow/sklearn-diabetes/src/score.py (hunk 3 above)
Comment (santiagxf) [reply sdk/python/endpoints/online/mlflow/sklearn-diabetes/src/score.py]: Addressed

**Review by dhmanigit — APPROVED** [review]

**Pull request merged**